src/lightning/data/streaming/dataloader.py

```diff
@@ -370,21 +370,20 @@ def __iter__(self) -> Any:
         else:
             yield from super().__iter__()
 
-    def state_dict(self) -> Optional[Dict[str, Any]]:
```

Comment:
@tchaton We need to follow the Stateful protocol for the state dict. See `lightning.fabric.utilities.types._Stateful`. It would be quite challenging to work with something that is not a dict here. Could we try to make it work?

Reply:
Yes, for sure. I was just experimenting with an idea. Let me move to dict.